src/Raven.Server/Documents/Indexes/Static/IndexCompiler.cs

```diff
@@ -533,6 +533,9 @@ private static StatementSyntax HandleReduce(string reduce, FieldNamesValidator f
                         results.Add(RoslynHelper.This(nameof(StaticTimeSeriesIndexBase.AddReferencedCollection)).Invoke(collection, rc).AsExpressionStatement());
                     }
                 }
+
+                if (mapRewriter.HasLoadCompareExchangeValue)
+                    results.Add(RoslynHelper.This(nameof(StaticIndexBase.AddCompareExchangeReferenceToCollection)).Invoke(collection).AsExpressionStatement());
```

Comment:
Why is this relevant at the collection level?

Reply:
We need to know which collections have compare exchange references. This is used in a lot of places e.g. during etag calculation